src/main/webapp/index-script.js

```diff
@@ -0,0 +1,112 @@
+  const searchBar = document.querySelector(".search");
+  const logInButton = document.querySelector("#logIn");
+
+  // Make 'clear-icon' visible when user starts typing.
+  searchBar.addEventListener("keyup", () => {
+    if(searchBar.value && clearIcon.style.visibility != "visible"){
+      clearIcon.style.visibility = "visible";
+    } else if(!searchBar.value) {
+      clearIcon.style.visibility = "hidden";
+    }
+  });
+
+  // Delete text and hide 'clear-icon' on click.
+  clearIcon.addEventListener("click", () => {
+    searchBar.value = "";
+    clearIcon.style.visibility = "hidden";
+  });
+
+  searchIcon.addEventListener("click", () => {
+    return; // TO-DO: Send search bar info to places API.
+  });
+
+  logInButton.addEventListener("click", () => {
+    return; // TO-DO: Fetch log-in info here.
+  });
+}
+
+/* Creates a map centered at the Googleplex! .*/
+function createMap() {
+  const googleplex = {lat: 37.422, lng: -122.0841};
+  map = new google.maps.Map(
+    document.getElementById('map'),
+    {center: googleplex, zoom: 13,
+    mapTypeControlOptions: {mapTypeIds: ['roadmap']}});
+
+  // Create the initial InfoWindow.
+  var infoWindow = new google.maps.InfoWindow(
+      {content: 'Open javascript console (ctrl + shift + j) then click the map to see the placeIDs of nearby locations (within 50m)',
+       position: googleplex});
+  infoWindow.open(map);
+
+  // Search by coordinates on map click.
+  map.addListener('click', function(mapsMouseEvent) {
+    infoWindow.close();
+    searchByCoordinates(mapsMouseEvent.latLng);
+  });
+}
+
+/* Takes coordinates from mouseclick and searches for locations within a given radius of those coordinates. */
+function searchByCoordinates(coordinate) {
+  var request = {
+    location: coordinate,
+    radius: '50',
+    rankBy: google.maps.places.RankBy.PROMINENCE
+  };
+
+  var service = new google.maps.places.PlacesService(map);
+  // Print resulting IDs to log.
+  service.nearbySearch(request, (results, status) => {
+    if (status == google.maps.places.PlacesServiceStatus.OK) {
+      results.forEach((result)=> {
+        var request = {
+          placeId: result.place_id,
+          fields: ['name']
+        };
+        service.getDetails(request, (place, status) => {
+          if (status == google.maps.places.PlacesServiceStatus.OK) {
+            console.log(place.name);
+          }
+        })
+      });
+    }
+  });
+}
+
+// Search using text query
+function searchByText(){
+  // Create the places service.
+  const service = new google.maps.places.PlacesService(map);
+  
+  // Perform a query (hard-coded to be the Googleplex for right now)
+  var request = {
+      query: 'Googleplex',
+      radius: '1000',
+      fields: ['place_id', 'geometry']
+  }
+
+  service.findPlaceFromQuery(request, function(results, status) {
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      for (var i = 0; i < results.length; i++){
+          // Populate the screen
```

Comment:
Modal Structure would work here, I will do implementation for next PR.

Reply:
For function that's incomplete, consider adding "TODO" at the front of the comment.